File: c/605.种花问题.c
/*
 * @lc app=leetcode.cn id=605 lang=c
 *
 * [605] 种花问题
 */

// @lc code=start
#include <stdbool.h>
bool canPlaceFlowers(int* flowerbed, int flowerbedSize, int n) {
    int count = 0;
    
    for(int i = 0; i < flowerbedSize; i++)
    {
        if (flowerbed[i] == 0)
        {
            bool left_ok = (i == 0) || (flowerbed[i - 1] == 0);
            bool right_ok = (i == flowerbedSize - 1) || (flowerbed[i + 1] == 0);

            if (left_ok && right_ok)
            {
                flowerbed[i] = 1;
                count++;
                if (count >= n)
                {
                    return 1;
                }
            }
        }
    }
    return count >= n;
}
// @lc code=end

File: python3/605.种花问题.py
#
# @lc app=leetcode.cn id=605 lang=python3
#
# [605] 种花问题
#

# @lc code=start
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        count = 0
        length = len(flowerbed)
        for i in range(length):
            if flowerbed[i] == 0:
                left_ok = (i == 0) or (flowerbed[i-1] == 0)
                right_ok = (i == length - 1) or (flowerbed[i+1] == 0)
                if left_ok and right_ok:
                    flowerbed[i] = 1
                    count += 1
                    if count >= n:
                        return True
        return count >= n
# @lc code=end

